### ISSUE TITLE
Feature Request: allow customizing warnings/errors

### ISSUE THREAD

**Issue by JasonMFry** [stil4m/elm-analyse#173]
Similar to [hlint](http://community.haskell.org/~ndm/darcs/hlint/hlint.htm#customization) in Haskell, it would be nice to be able to add custom warnings/errors. 

As an example, in our codebase we have a wrapper function around `onClick` that enforces adding an `id` to each clickable element. We want to make it very difficult to use `onClick` directly, so it would be nice if elm-analyse would complain if someone uses `onClick` outside of our helper module.

**Comment by akoppela** [comment]
Yes, in our company we have several custom linters. And it's always nice to have a way to write custom linters. At the moment we use Elm worker and `elm-syntax` to write linters.
